sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala

```diff
@@ -224,7 +224,8 @@ class SQLQueryTestSuite extends QueryTest with SharedSQLContext {
   private def runTest(testCase: TestCase): Unit = {
     val input = fileToString(new File(testCase.inputFile))
 
-    val (comments, code) = input.split("\n").partition(_.startsWith("--"))
+    val (comments, codeMaybeWithComment) = input.split("\n").partition(_.trim.startsWith("--"))
```

Comment:
Add `trim` to filter out these comments:
https://github.com/apache/spark/blob/d440699192f21b14dfb8ec0dc5673537e1003b55/sql/core/src/test/resources/sql-tests/inputs/subquery/in-subquery/not-in-unit-tests-multi-column.sql#L32-L33